src/utils/parse_utils.ts

```diff
@@ -41,85 +44,82 @@ export const parseUtils = {
+            if (request.takerAddress === undefined) {
+                throw new ValidationError([
+                    {
+                        field: 'takerAddress',
+                        code: ValidationErrorCodes.RequiredField,
+                        reason: ValidationErrorReasons.TakerAddressInvalid,
+                    },
+                ]);
+            }
 
-                // If the user is requesting a firm quote, we want to make sure that `intentOnFilling` is set to "true".
-                if (endpoint === 'quote' && request.intentOnFilling !== 'true') {
-                    throw new ValidationError([
-                        {
-                            field: 'intentOnFilling',
-                            code: ValidationErrorCodes.RequiredField,
-                            reason: ValidationErrorReasons.RequiresIntentOnFilling,
-                        },
-                    ]);
-                }
+            // We enforce a valid API key - we don't want to fail silently.
+            if (request.apiKey === undefined) {
+                throw new ValidationError([
+                    {
+                        field: '0x-api-key',
+                        code: ValidationErrorCodes.RequiredField,
+                        reason: ValidationErrorReasons.InvalidApiKey,
+                    },
+                ]);
+            }
+            if (!validApiKeys.includes(request.apiKey)) {
+                throw new ValidationError([
+                    {
+                        field: '0x-api-key',
+                        code: ValidationErrorCodes.FieldInvalid,
+                        reason: ValidationErrorReasons.InvalidApiKey,
+                    },
+                ]);
+            }
 
-                return {
-                    nativeExclusivelyRFQT: true,
-                    excludedSources: Object.values(ERC20BridgeSource).filter(t => t !== ERC20BridgeSource.Native),
-                };
-            } else {
+            // If the user is requesting a firm quote, we want to make sure that `intentOnFilling` is set to "true".
+            if (endpoint === 'quote' && request.intentOnFilling !== 'true') {
                 throw new ValidationError([
                     {
-                        field: 'includedSources',
-                        code: ValidationErrorCodes.IncorrectFormat,
-                        reason: ValidationErrorReasons.ArgumentNotYetSupported,
+                        field: 'intentOnFilling',
+                        code: ValidationErrorCodes.RequiredField,
+                        reason: ValidationErrorReasons.RequiresIntentOnFilling,
                     },
                 ]);
             }
+
+            return {
+                excludedSources: [],
+                includedSources: parseUtils.parseStringArrForERC20BridgeSources(['0x', ...includedIds]),
+                nativeExclusivelyRFQT: !includedIds.includes('0x') && !includedIds.includes('Native'),
```

Comment:
Could we simply make RFQT a special edge case here? for example, as of today there is no taker that requests exclusive RFQT liquidity alongside another on-chain source. If we feel good about making this assumption, we can add the following checks and statements in the code:

- If `RFQT` is added to `includedSources`, then `includedIds.length` must == 1
- `nativeExclusivelyRFQT: includedIds.includes("RFQT")`. So you could keep `L104` as `nativeExclusivelyRFQT: true`.

Reply:
Idk, I could see simbot playing with a mix of sources and excluding open orderbook. The behavior for a taker passing in just `includedSources=RFQT` is still the same, right? What is the motivation for not allowing other included sources to coexist with RFQT?